src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs

```diff
@@ -364,12 +378,185 @@ ref Unsafe.Add(ref valueRef, 1), valueTailLength))
+                    // For each set bits, we'll check if it's a match at that location.
+                    uint mask = cmpAnd.ExtractMostSignificantBits();
+                    do
+                    {
+                        // Do a full IgnoreCase equality comparison. SpanHelpers.IndexOf skips comparing the two characters in some cases,
+                        // but we don't actually know that the two characters are equal, since we compared with | 0x20. So we just compare
+                        // the full string always.
+                        int bitPos = BitOperations.TrailingZeroCount(mask);
+                        nint charPos = (nint)((uint)bitPos / 2); // div by 2 (shr) because we work with 2-byte chars
+                        if (EqualsIgnoreCase(ref Unsafe.Add(ref searchSpace, offset + charPos), ref valueRef, value.Length))
+                        {
+                            // Match! Return the index.
+                            return (int)(offset + charPos);
+                        }
+
+                        // Clear the two lowest set bits in the mask. If there are no more set bits, we're done.
+                        // If any remain, we loop around to do the next comparison.
+                        if (Bmi1.IsSupported)
+                        {
+                            mask = Bmi1.ResetLowestSetBit(Bmi1.ResetLowestSetBit(mask));
+                        }
+                        else
+                        {
+                            mask &= ~(uint)(0b11 << bitPos);
+                        }
+                    } while (mask != 0);
+                    goto LoopFooter;
+
+                } while (true);
+            }
+            else // 128bit vector path (SSE2 or AdvSimd)
+            {
+                // Create a vector for each of the lowercase ASCII characters we're searching for.
+                Vector128<ushort> ch1 = Vector128.Create((ushort)valueChar);
+                Vector128<ushort> ch2 = Vector128.Create((ushort)valueCharU);
+
+                nint searchSpaceMinusValueTailLengthAndVector = searchSpaceMinusValueTailLength - (nint)Vector128<ushort>.Count;
+                do
+                {
+                    // Make sure we don't go out of bounds.
+                    Debug.Assert(offset + ch1ch2Distance + Vector128<ushort>.Count <= source.Length);
+
+                    // Load a vector from the current search space offset and another from the offset plus the distance between the two characters.
+                    // For each, | with 0x20 so that letters are lowercased, then & those together to get a mask. If the mask is all zeros, there
+                    // was no match.  If it wasn't, we have to do more work to check for a match.
+                    Vector128<ushort> cmpCh2 = Vector128.Equals(ch2, Vector128.BitwiseOr(Vector128.LoadUnsafe(ref searchSpace, (nuint)(offset + ch1ch2Distance)), Vector128.Create((ushort)0x20)));
```

Comment:
Very nit: `Vector128.BitwiseOr` -> ` | `

Reply:
This is just style, right? Happy to change it, just questioning whether it's worth rerunning ci.

Reply:
definitely not worth it 🙂

Reply:
Right its "just style". There is also the general considerations of "methods" vs "operators" (such as precedence and readability) but we're not super consistent today just due to the operators being relatively new.